servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -188,18 +189,44 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge) thro
   }
 
   @Override
-  public EntriesResponse getEntries(String refName) throws NessieNotFoundException {
+  public EntriesResponse getEntries(String refName, List<String> valueTypes) throws NessieNotFoundException {
     final Hash hash = getHashOrThrow(refName);
+
+    final Set<Type> valueTypeSet = valueTypes.stream().map(Type::valueOf).collect(Collectors.toSet());
     try {
       List<EntriesResponse.Entry> entries = getStore().getKeys(hash)
-          .map(key -> EntriesResponse.Entry.builder().name(fromKey(key)).type(Type.UNKNOWN).build())
+          .map(key -> EntriesResponse.Entry.builder().name(fromKey(key.getEntity())).type(toType(key.getEntityType())).build())
+          .filter(x -> valueTypeSet.isEmpty() || valueTypeSet.contains(x.getType()))
           .collect(ImmutableList.toImmutableList());
       return EntriesResponse.builder().addAllEntries(entries).build();
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(String.format("Unable to find the reference [%s].", refName), e);
     }
   }
 
+
+  private static Type toType(String valueType) {
+    switch (valueType) {
+      case "IcebergTable":
+      case "ImmutableIcebergTable":
```

Comment:
That seems wrong that 2 different value types point to the same Type object

Reply:
agreed, I think based on conversations on #834 we will likely change this